### DISCUSSION TITLE
I'm trying to create a broad 'OR' query with multiple AND clauses on each branch

### DISCUSSION THREAD

**Discussion by mshiltonj** [activerecord-hackery/ransack#1409] [Q&A]
I have a search that looks like:

    search = {
      name: 'bob',
      updated_at_gt: '2023-01-01'
    }

Which has an sql query that looks something like this:

    ... things.name = 'bob' AND things.updated_at > '2023-01-01' ...

We have 'draft' association, and I'm now transforming the incoming search to look like:

    search= {
      name_or_draft_name: 'bob'
      updated_at_or_draft_updated_at_gt > '2023-01-01'
    }

And the sql query looks like this:

    ... (things.name='bob' OR drafts.name='bob') AND (things.updated_at > '2023-01-01' OR drafts.updated_at > '2023-01-01') ...

Which makes sense and I understanding what's going on. But what I'm do is have have a query that looks like this:

    ... (things.name = 'bob' AND things.updated_at > '2023-01-01') OR (drafts.name = 'bob' AND drafts.updated_at > '2023-01-01') ...
    
I can manipulate the search params at will, but I'm not seeing how or if I can, with ransack, build an OR query where each branch has multiple AND clauses. Is something like this possible in ransack?

**Comment by reid-rigo** [comment]
I believe this can be achieved with explicit groupings:

```ruby
Model.ransack(
  m: "or", 
  g: [
    { name: 'bob', updated_at_gt: '2023-01-01' }, 
    { name_or_draft_name: 'bob', updated_at_or_draft_updated_at_gt > '2023-01-01' } 
  ]
)
```

**Comment by scarroll32** [comment]
Closing as answered
